### ISSUE TITLE
Jepsen bank/move-tablet test causes Alpha panic

### ISSUE THREAD

**Issue by danielmai** [dgraph-io/dgraph#3267]
If you suspect this could be a bug, follow the template.

- What version of Dgraph are you using?
master only (**not** latest v1.0.13 or v1.0.14-rc2).

```
$ dgraph version

Dgraph version   : v1.0.12-rc3-289-g39560db81
Commit SHA-1     : 39560db81
Commit timestamp : 2019-04-05 15:38:35 -0700
Branch           : master
Go version       : go1.11.5
```

- Steps to reproduce the issue (command/config used to run Dgraph).

- Run Jepsen bank test with move tablet (run in control node)

```
lein run test --workload bank --nemesis move-tablet --time-limit 600 --concurrency 6n --rebalance-interval 10h --local-binary /gobin/dgraph
```

If `:unavailable` starts showing up in the Jepsen log, then look at the Alpha logs in each of the five nodes. e.g., when I ran it I checked the fourth node's Alpha log:

```
$ docker exec jepsen-n4 tail -f /opt/dgraph/alpha.log
I0408 17:46:50.534559   31642 predicate_move.go:62] Receiving predicate. Batching and proposing key values
I0408 17:46:50.534583   31642 predicate_move.go:78] Predicate being received: key_0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1181bbf]

goroutine 3646 [running]:
github.com/dgraph-io/dgraph/query.ProcessGraph(0x16126a0, 0xc00bb09530, 0xc00d03fb00, 0xc00d03f200, 0xc00b7b5ec0)
	/home/dmai/go/src/github.com/dgraph-io/dgraph/query/query.go:2074 +0x16ff
created by github.com/dgraph-io/dgraph/query.ProcessGraph
	/home/dmai/go/src/github.com/dgraph-io/dgraph/query/query.go:2239 +0xc90
```

- Expected behaviour and actual result.

All Dgraph nodes should continue running during the test.

**Comment by danielmai** [comment]
Also saw this index error:

```
Dgraph version   : v1.0.12-rc3-340-gb1d750b0c
Commit SHA-1     : b1d750b0c
Commit timestamp : 2019-04-23 14:52:36 -0700
Branch           : master
Go version       : go1.11.5
```

```
I0424 00:26:49.378077     395 predicate_move.go:182] Was instructed to delete tablet: key_1
I0424 00:26:49.379283     395 server.go:529] Got a query: query:"query all($type: string) { q(func: eq(type_0, $type)) {\nkey_0\nkey_1\nkey_2\namount_0\namount_1\namount_2}}" vars:<key:"$type" value:"account" > lin_read:<> 
I0424 00:26:49.379384     395 server.go:529] Got a query: query:"query all($key: int) { q(func: eq(key_2, $key)) { uid key_2 amount_2 } } " vars:<key:"$key" value:"5" > start_ts:1130 lin_read:<> 
I0424 00:26:49.446226     395 server.go:529] Got a query: query:"query all($type: string) { q(func: eq(type_1, $type)) {\nkey_0\nkey_1\nkey_2\namount_0\namount_1\namount_2}}" vars:<key:"$type" value:"account" > start_ts:1139 lin_read:<> 
I0424 00:26:49.448592     395 server.go:529] Got a query: query:"query all($type: string) { q(func: eq(type_2, $type)) {\nkey_0\nkey_1\nkey_2\namount_0\namount_1\namount_2}}" vars:<key:"$type" value:"account" > start_ts:1139 lin_read:<> 
I0424 00:26:49.457570     395 index.go:961] Dropping predicate: [key_1]
I0424 00:26:49.457610     395 node.go:85] DropPrefix called on 00000000  00 00 05 6b 65 79 5f 31                           |...key_1|
. Blocking writes...
I0424 00:26:49.457630     395 node.go:85] Writes flushed. Stopping compactions now...
I0424 00:26:49.457694     395 groups.go:866] Batched 7 updates. Max Assigned: 1153. Proposing Deltas:
I0424 00:26:49.459540     395 server.go:529] Got a query: query:"query all($key: int) { q(func: eq(key_1, $key)) { uid key_1 amount_1 } } " vars:<key:"$key" value:"2" > start_ts:1147 lin_read:<> 
I0424 00:26:49.546107     395 node.go:85] Got compaction priority: {level:0 score:1.74 dropPrefix:[0 0 5 107 101 121 95 49]}
I0424 00:26:49.546166     395 node.go:85] Running for level: 0
I0424 00:26:49.610721     395 node.go:85] LOG Compact 0->1, del 2 tables, add 1 tables, took 64.513166ms
I0424 00:26:49.610893     395 node.go:85] Compaction for level: 0 DONE
I0424 00:26:49.655523     395 groups.go:262] group 2 checksum: 13590400285836887974
I0424 00:26:49.663206     395 node.go:85] LOG Compact 1->1, del 1 tables, add 1 tables, took 52.235406ms
I0424 00:26:49.663465     395 node.go:85] DropPrefix done
I0424 00:26:49.663498     395 node.go:85] Resuming writes
I0424 00:26:49.663547     395 schema.go:77] Deleting schema for predicate: [key_1]
I0424 00:26:49.666361     395 server.go:529] Got a query: query:"query all($key: int) { q(func: eq(key_1, $key)) { uid key_1 amount_1 } } " vars:<key:"$key" value:"3" > lin_read:<> 
2019/04/24 00:26:49 Attribute key_1 is not indexed.

github.com/dgraph-io/dgraph/x.Wrap
        /home/dmai/go/src/github.com/dgraph-io/dgraph/x/error.go:100
github.com/dgraph-io/dgraph/x.Check
        /home/dmai/go/src/github.com/dgraph-io/dgraph/x/error.go:42
github.com/dgraph-io/dgraph/worker.(*queryState).handleCompareFunction
        /home/dmai/go/src/github.com/dgraph-io/dgraph/worker/task.go:1034
github.com/dgraph-io/dgraph/worker.(*queryState).helpProcessTask
        /home/dmai/go/src/github.com/dgraph-io/dgraph/worker/task.go:867
github.com/dgraph-io/dgraph/worker.processTask
        /home/dmai/go/src/github.com/dgraph-io/dgraph/worker/task.go:755
github.com/dgraph-io/dgraph/worker.(*grpcWorker).ServeTask.func1
        /home/dmai/go/src/github.com/dgraph-io/dgraph/worker/task.go:1655
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1333
2019/04/24 00:26:49 Attribute key_1 is not indexed.

github.com/dgraph-io/dgraph/x.Wrap
        /home/dmai/go/src/github.com/dgraph-io/dgraph/x/error.go:100
github.com/dgraph-io/dgraph/x.Check
        /home/dmai/go/src/github.com/dgraph-io/dgraph/x/error.go:42
github.com/dgraph-io/dgraph/worker.(*queryState).handleCompareFunction
        /home/dmai/go/src/github.com/dgraph-io/dgraph/worker/task.go:1034
github.com/dgraph-io/dgraph/worker.(*queryState).helpProcessTask
        /home/dmai/go/src/github.com/dgraph-io/dgraph/worker/task.go:867
github.com/dgraph-io/dgraph/worker.processTask
        /home/dmai/go/src/github.com/dgraph-io/dgraph/worker/task.go:755
github.com/dgraph-io/dgraph/worker.(*grpcWorker).ServeTask.func1
        /home/dmai/go/src/github.com/dgraph-io/dgraph/worker/task.go:1655
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1333
```

**Comment by danielmai** [comment]
The first panic can reproduced by running `dgraph increment` on an Alpha and the continually moving the counter predicate to different groups.

Increment tool:
```
dgraph increment --addr localhost:9180 --pred=counter.val --num=10000
```

Moving tablets continually:
```
#!/bin/bash
i=0
g=3 # number of groups
for i in $(seq 1 10000); do
    i=$((i%g+1))
    curl "localhost:6180/moveTablet?tablet=counter.val&group=$i"
    echo
    sleep 0.1
done
```

```
$ git bisect log
# bad: [24fd75992fe7da040f26bde08a2c14dc639d1624] Use jsonpb library to unmarshal alter HTTP requests. (#3314)
# good: [f7547e48fce76af1e2c390dcee8b4e771c12fd47] Switch all node ids to hex, so they are consistent with Raft logs.
git bisect start 'master' 'f7547e48fce76af1e2c390dcee8b4e771c12fd47'
# good: [f124e5663a1354197d99e62a54a673dfc8b68e33] query/query.go: increase verbosity level of missing value warning. (#3092)
git bisect good f124e5663a1354197d99e62a54a673dfc8b68e33
# bad: [94a9b6cad6ea238266f5319138b62fc1cc8a584e] Change query tests to use standard lang RDF syntax (#3215)
git bisect bad 94a9b6cad6ea238266f5319138b62fc1cc8a584e
# good: [6398f85bddceb74dc933276d4484670d9819f27f] Fix bugs related to best-effort queries (#3125)
git bisect good 6398f85bddceb74dc933276d4484670d9819f27f
# bad: [a57dfc0ca948dfadd54795f95c25edf7b5045b66] Stream Raft Messages and Fix Check Quorum (#3138)
git bisect bad a57dfc0ca948dfadd54795f95c25edf7b5045b66
# bad: [351d6051b58a3345591e4cba5c29e9841156b47d] Fix the aclCache race condition by initializing it on definition (#3141)
git bisect bad 351d6051b58a3345591e4cba5c29e9841156b47d
# good: [17435efca070ad398b9adf37e13a9e29f5b89de6] Support comments in schema (#3133)
git bisect good 17435efca070ad398b9adf37e13a9e29f5b89de6
# good: [9a8dbd122bc5b72ceae858bdaa6855fdfcadaf85] Reduce test time (#3134)
git bisect good 9a8dbd122bc5b72ceae858bdaa6855fdfcadaf85
# bad: [ef74ed80da6c4c11b27e4ff1b91f8f3d9b4b2864] Prevent alphas from asking zero to serve tablets during queries. (#3091)
git bisect bad ef74ed80da6c4c11b27e4ff1b91f8f3d9b4b2864
# good: [abc727e2fe01db3c4dc6da50728819c377d36b3e] Ensure reserved predicates cannot be moved. (#3137)
git bisect good abc727e2fe01db3c4dc6da50728819c377d36b3e
# first bad commit: [ef74ed80da6c4c11b27e4ff1b91f8f3d9b4b2864] Prevent alphas from asking zero to serve tablets during queries. (#3091)
```

Git bisect narrowed down the change to ef74ed80da6c4c11b27e4ff1b91f8f3d9b4b2864.

**Comment by martinmr** [comment]
The original panic has been fixed by #3318

**Comment by martinmr** [comment]
Closing this issue as the panic has been fixed.
